packages/ui-extensions/src/surfaces/checkout/preact/subscription.ts

```diff
@@ -0,0 +1,45 @@
+import {useEffect, useState} from 'preact/hooks';
+import type {StatefulRemoteSubscribable} from '@remote-ui/async-subscription';
+
+type Subscriber<T> = Parameters<StatefulRemoteSubscribable<T>['subscribe']>[0];
+
+/**
+ * Subscribes to the special wrapper type that all "changeable" values in the
+ * checkout use. This hook extracts the most recent value from that object,
+ * and subscribes to update the value when changes occur in the checkout.
+ *
+ * > Note:
+ * > You generally shouldn't need to use this directly, as there are dedicated hooks
+ * > for accessing the current value of each individual resource in the checkout.
+ */
+export function useSubscription<Value>(
+  subscription: StatefulRemoteSubscribable<Value>,
```

Comment:
I'm actually unsure why we need to keep the `StatefulRemoteSubscribable` at all. I thought we were also planning to send over the data with the Signal-light interface with `value` and `subscribe`. Couldn't we use the new interface?

Reply:
The intention here is to preserve the exact nature of how these hooks work today to reduce the amount of code existing extensions will need to re-write. But, they will only remain for backwards compatibility. We will move towards a signal approach and document those as best practices going forward. We won't have that in time for the RC though, but should have that before July.